### PR TITLE
updating some details in pathfinding spec, adding high level description

### PR DESCRIPTION
**Pull request by err508**

### CODE REVIEW

**Comment by palango** [comment]
Most comments from #42 still apply.

**Comment by err508** [comment]
commit --amend didn't go through I think.

File: pathfinding_service.rst [hunk 1]
```diff
@@ -119,21 +133,27 @@ Example
...
 | Field Name           | Field Type    |  Description                                                          |
 +======================+===============+=======================================================================+
 | channel_id           | int           | The channel for which the fee should be updated.                      |
 +----------------------+---------------+-----------------------------------------------------------------------+
+|nonce                 | int           | A nonce for replay protection.                                        |
++----------------------+---------------+-----------------------------------------------------------------------+
 | fee                  | int           | The new fee to be set.                                                |
 +----------------------+---------------+-----------------------------------------------------------------------+
-| signature            | bytes         | The signature of the channel partner for whom the channel is outgoing.|
+| signature            | bytes         | signature of a channel partner                                        |
```
Comment (palango) [inline pathfinding_service.rst]: big s in signature

File: pathfinding_service.rst [hunk 2]
```diff
@@ -119,21 +133,27 @@ Example
     }
 
 
-``api/1/channels/<channel_id>/fee``
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+``api/1/channels/<channel_id>/update_fee``
```
Comment (palango) [inline pathfinding_service.rst]: should just be fee for the same reasons as `path`

File: pathfinding_service.rst [hunk 3]
```diff
@@ -164,10 +185,14 @@ Example
 ``api/1/paths``
 ^^^^^^^^^^^^^^^
 
-Request a list of possible paths from startpoint to endpoint for a given transfer value.
+The method will do ``num_paths`` iterations of Dijkstra on the last-known state of the Raiden
+Network (regarded as directed weighted graph) to return ``num_paths`` different paths for a mediated transfer of ``value``.
+
+* checks if an edge (i.e. a channel) has ``capacity > value``, else ignores it.
```
Comment (palango) [inline pathfinding_service.rst]: I prefer capitalization in lists

File: pathfinding_service.rst [hunk 4]
```diff
@@ -119,21 +133,27 @@ Example
...
-``api/1/channels/<channel_id>/fee``
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+``api/1/channels/<channel_id>/update_fee``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 Update the fee for the given channel, for the outgoing channel from the partner who signed the message.
+A nonce is required to be incorporated in the signature for replay protection.
+
+* Reconstructs the signers ``public_key`` of a requested fee update with coincurve's ``from_signature_and_message`` method.
+
+* Derives the two ``channel_participants`` with ``from channel_id``. Checks if the signing ``public_key`` matches one of the ``channel participant``'s ``address``, else returns respectively.
```
Comment (palango) [inline pathfinding_service.rst]: returns an error if the signature doesn't match

File: pathfinding_service.rst [hunk 5]
```diff
@@ -164,10 +185,14 @@ Example
 ``api/1/paths``
 ^^^^^^^^^^^^^^^
 
-Request a list of possible paths from startpoint to endpoint for a given transfer value.
+The method will do ``num_paths`` iterations of Dijkstra on the last-known state of the Raiden
```
Comment (palango) [inline pathfinding_service.rst]: Dijkstras algorithm

File: pathfinding_service.rst [hunk 6]
```diff
@@ -164,10 +185,14 @@ Example
...
+Network (regarded as directed weighted graph) to return ``num_paths`` different paths for a mediated transfer of ``value``.
+
+* checks if an edge (i.e. a channel) has ``capacity > value``, else ignores it.
 
-This method will be rate-limited in a configurable way. If the rate limit is exceeded, clients can be required to pay the path-finding service with RDN tokens via the Raiden Network. The required path for this payment will be provided by the service for free. This enables a simple user experience for light users without the need for additional on-chain transactions for channel creations or payments, while at the same time monetizing extensive use of the API.
-To get payment information the *get_payment_info* method is used.
+* applies on the fly changes to the graph's weights - depends on ``DIVERSITY_PEN_DEFAULT`` from ``config``, to penalize edges which are part of a path that is returned already.
+
+* depends on a user preference via the ``bias`` argument, to decided the trade off between fee-level vs path-length (i.e. cost vs speed) - default ``bias = 0``, i.e. full fee minimization.
```
Comment (palango) [inline pathfinding_service.rst]: point after vs.

File: pathfinding_service.rst [hunk 7]
```diff
@@ -164,10 +185,14 @@ Example
...
 
-Request a list of possible paths from startpoint to endpoint for a given transfer value.
+The method will do ``num_paths`` iterations of Dijkstra on the last-known state of the Raiden
+Network (regarded as directed weighted graph) to return ``num_paths`` different paths for a mediated transfer of ``value``.
+
+* checks if an edge (i.e. a channel) has ``capacity > value``, else ignores it.
 
-This method will be rate-limited in a configurable way. If the rate limit is exceeded, clients can be required to pay the path-finding service with RDN tokens via the Raiden Network. The required path for this payment will be provided by the service for free. This enables a simple user experience for light users without the need for additional on-chain transactions for channel creations or payments, while at the same time monetizing extensive use of the API.
-To get payment information the *get_payment_info* method is used.
+* applies on the fly changes to the graph's weights - depends on ``DIVERSITY_PEN_DEFAULT`` from ``config``, to penalize edges which are part of a path that is returned already.
```
Comment (palango) [inline pathfinding_service.rst]: Capitalize in lists

File: pathfinding_service.rst [hunk 8]
```diff
@@ -133,11 +152,13 @@ Arguments
 +======================+===============+=======================================================================+
 | token_network_address| address       | The token network address for which the payment info is requested.    |
 +----------------------+---------------+-----------------------------------------------------------------------+
-| channel_id           | int           | The channel for which the fee should be updated.                      |
+| Channel_id           | int           | The channel for which the fee should be updated.                      |
```
Comment (palango) [inline pathfinding_service.rst]: These shouldn't be capitalized

**Pull request merged**